### ISSUE TITLE
🐛 Bug Report:  Delayed digested flow digests but sends notification multiple times

### ISSUE THREAD

**Issue by sebastiaanviaene** [novuhq/novu#1633]
### 📜 Description

My workflow as follows:

- I delay the messages with a scheduledAt parameter to wait until a certain time (tomorrow 8AM for example)
- I then put a digest block to combine those delayed messages into a "Daily update mail" for 10 seconds (assuming the workflow won't take much time to send them to the next block)

It seems if I am sending 2 messages like this, it sends me 2 digested notifications (with the same content) instead of 1 

### 👟 Reproduction steps

Make workflow like this:

<img width="332" alt="image" src="https://user-images.githubusercontent.com/6652818/195109006-7d1e9b71-fa76-4e77-b3e9-fd52032eddf3.png">

<img width="334" alt="image" src="https://user-images.githubusercontent.com/6652818/195109930-9cc32896-cd65-471b-82dd-89d98907ad8a.png">
<img width="358" alt="image" src="https://user-images.githubusercontent.com/6652818/195109939-54b80419-6166-416e-9c07-a54382552904.png">

With this content in the in-app notification:

```
Total events in digest: {{step.total_count}}
{{#if step.digest}}
{{#each step.events}}
    {{test}}
 {{/each}}
 {{else}}
 Not a digested template
 {{/if}}
```

Send 2 events like this:

```json
{
	"name": "test-digest-delay",
    "to": {
        "subscriberId": "3",
        "email": "sebviaene@gmail.com"
    },
    "payload": {
    	"scheduledAt": "2022-10-11T13:42:00.992Z",
    	"test": "1"
    }
}
```

### 👍 Expected behavior

It should digest and send me one notification

### 👎 Actual Behavior with Screenshots

it sends 2 digested notifications

<img width="394" alt="image" src="https://user-images.githubusercontent.com/6652818/195108889-44bdc419-51d8-4c04-84bc-dc949ad9d4fd.png">

### 💻 Operating system

MacOS

### 🤖 Node Version

Stable (16.x)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by scopsy** [comment]
@ainouzgali could you please take a look at this one? I don't see a reason of why it might not work.

**Comment by ainouzgali** [comment]
Yes, I will look into it.

**Comment by sebastiaanviaene** [comment]
@ainouzgali any luck?

**Comment by sebastiaanviaene** [comment]
Were you able to reproduce it?

**Comment by ainouzgali** [comment]
Yes, I've been working on it @sebastiaanviaene. But it's a bit tricky because of the way our queue works.  
 It's a very specific bug of Digest coming after a Scheduled Delay that has been triggered more than once for the exact same scheduled date.
 If you need a workaround just for now, you could try scheduling the sendAt parameter for 1 or 2 seconds more than the sendAt of your first trigger.  
 For example - if your first trigger has  `"sendAt": "2022-10-12T17:52:15.694Z"` then for the next triggers try using `"sendAt": "2022-10-12T17:54:15.694Z"`
 
But I am working on a fix - I will keep you updated  :)

**Comment by sebastiaanviaene** [comment]
@ainouzgali Thank you for the workaround and looking into it! Let me know if I can help in any way :)

**Comment by ainouzgali** [comment]
Hi @sebastiaanviaene, the fix was merged to main branch. This will be included in the closest release (v0.9.0).
